modules/imgproc/src/color.cpp

```diff
@@ -6767,7 +6767,10 @@ struct Lab2RGBinteger
 
         //float fxz[] = { ai / 500.0f + fy, fy - bi / 200.0f };
         int adiv, bdiv;
-        adiv = aa*BASE/500 - 128*BASE/500, bdiv = bb*BASE/200 - 128*BASE/200;
+        //adiv = aa*BASE/500 - 128*BASE/500, bdiv = bb*BASE/200 - 128*BASE/200;
+        //approximations with reasonable precision
+        adiv = ((5*aa*53687 + (1 << 7)) >> 13) - 128*BASE/500;
+        bdiv = ((  bb*41943 + (1 << 4)) >>  9) - 128*BASE/200+1;
```

Comment:
Looks like these code lines are not equal to their comments.

`adiv = aa*BASE/500 - 128*BASE/500`  ==> `adiv = (aa-128)*BASE/500` ?

Reply:
The goal was to make non-vectorized code produce the same values as vectorized one.
This code is an approximation of the calculation which is done w/o integer division operation since there's no integer division available in SIMD.